### ISSUE TITLE
Error running maxquant_mqpar on Galaxy Australia

### ISSUE THREAD

**Issue by cat-bro** [galaxyproteomics/tools-galaxyp#544]
I’m not 100% sure, but I think that an error with maxquant_mqpar on Galaxy Australia is due to mqpar being used as an input name and an output name.
https://github.com/galaxyproteomics/tools-galaxyp/blob/master/tools/maxquant_mqpar/maxquant_mqpar.xml#L105
https://github.com/galaxyproteomics/tools-galaxyp/blob/master/tools/maxquant/macros.xml#L159

I cannot reproduce the error on Galaxy Europe.

The command looks like so:

```
ln -s '/mnt/user-data/003/422/dataset_3422791.dat' 'ABCDE.mzML' &&  python3 '/mnt/tools-indices/shed_tools/toolshed.g2.bx.psu.edu/repos/galaxyp/maxquant_mqpar/2d67fb758956/maxquant_mqpar/modify_mqpar.py' --infiles='ABCDE.mzML' --version=1.6.10.43 --num_threads=${GALAXY_SLOTS:-1} --substitution_rx='[^\w\-\s\.]' --fasta_files='/mnt/user-data/003/422/dataset_3422983.dat' --description_parse_rule='^>.*\|.*\|[^ ]+ (.*) OS.*$' --identifier_parse_rule='^>.*\|(.*)\|.*$' '/mnt/user-data/003/435/dataset_3435253.dat'  && maxquant mqpar.xml >> '/mnt/user-data/003/435/dataset_3435269.dat' && mv '/mnt/user-data/003/435/dataset_3435253.dat' combined/txt/mqpar.xml && tar -zcf '/mnt/user-data/003/435/dataset_3435268.dat' ./combined/txt
```

The input `‘$mqpar’` parameter ought to correspond to `'/mnt/user-data/003/435/dataset_3435253.dat'` in the above command, but this is not the input mqpar.xml file path, instead it is an empty file created at runtime, so an error is thrown from xml.etree trying to parse an empty file.

The correct input file is used only when mqpar.xml is excluded from the outputs.  The tool passes its test because the mqpar.xml output is not listed as an output for the test case.

The solution could be to either (a) rename the input parameter from mqpar to mqpar_input or (b) not have mqpar.xml as an output option.  I’ve tried (a) on a dev server and it works.

Moving the `'$mqpar'` file later in the command looks like a bug https://github.com/galaxyproteomics/tools-galaxyp/blob/master/tools/maxquant_mqpar/maxquant_mqpar.xml#L41 as it removes a user's dataset.  Should this be `mv mqpar.xml combined/txt/mqpar.xml`?

**Comment by bgruening** [comment]
Is this with pulsar in AU? I'm wondering why this does not happen in EU? To you have https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/config/sample/galaxy.yml.sample#L1937 activated?

**Comment by cat-bro** [comment]
Hi @bgruening, the tool is not running on pulsar.

The bug is replicated in the first commit on that PR.  The test report https://github.com/galaxyproteomics/tools-galaxyp/suites/2015761906/artifacts/40382986 shows the `xml.etree.ElementTree.ParseError` caused by trying to parse an empty file.

`$mqpar` is an ambiguous variable.  It could mean the input dataset or the output dataset depending on the order in which the variables are assigned.  It looks like the right dataset (the input dataset) is passed to modify_mqpar.py on Galaxy Europe but the wrong dataset (the output dataset) is passed to modify_mqpar.py on Galaxy Australia and also in the planemo test.  It could be that the choice of job runner affects this, but the bug is arising from the `$mqpar` having two meanings.

**Comment by cat-bro** [comment]
this was fixed for us with #559
